engine/userguide/storagedriver/aufs-driver.md

```diff
@@ -4,248 +4,222 @@ keywords: 'container, storage, driver, AUFS '
-mount point, must all exist on the same Linux host. AUFS refers to each
-directory that it stacks as a *branch*.
+## Configure Docker with the `aufs` storage driver
 
-Within Docker, AUFS union mounts enable image layering. The AUFS storage driver
- implements Docker image layers using this union mount system. AUFS branches
-correspond to Docker image layers. The diagram below shows a Docker container
-based on the `ubuntu:latest` image.
+If the AUFS driver is loaded into the kernel when you start Docker, and no other
+storage driver is configured, Docker uses it by default.
 
-![](images/aufs_layers.jpg)
+1.  Use the following command to verify that your kernel supports AUFS.
 
-This diagram shows that each image layer, and the container layer, is
-represented in the Docker hosts filesystem as a directory under
-`/var/lib/docker/`. The union mount point provides the unified view of all
-layers. As of Docker 1.10, image layer IDs do not correspond to the names of
-the directories that contain their data.
+    ```bash
+    $ grep aufs /proc/filesystems
 
-AUFS also supports the copy-on-write technology (CoW). Not all storage drivers
-do.
+    nodev   aufs
+    ```
 
-## Container reads and writes with AUFS
+2.  Check which storage driver Docker is using.
 
-Docker leverages AUFS CoW technology to enable image sharing and minimize the
-use of disk space. AUFS works at the file level. This means that all AUFS CoW
-operations copy entire files - even if only a small part of the file is being
-modified. This behavior can have a noticeable impact on container performance,
- especially if the files being copied are large, below a lot of image layers,
-or the CoW operation must search a deep directory tree.
+    ```bash
+    $ docker info
 
-Consider, for example, an application running in a container needs to add a
-single new value to a large key-value store (file). If this is the first time
-the file is modified, it does not yet exist in the container's top writable
-layer. So, the CoW must *copy up* the file from the underlying image. The AUFS
-storage driver searches each image layer for the file. The search order is from
- top to bottom. When it is found, the entire file is *copied up* to the
-container's top writable layer. From there, it can be opened and modified.
+    <truncated output>
+    Storage Driver: aufs
+     Root Dir: /var/lib/docker/aufs
+     Backing Filesystem: extfs
+     Dirs: 0
+     Dirperm1 Supported: true
+    <truncated output>
+    ```
 
-Larger files obviously take longer to *copy up* than smaller files, and files
-that exist in lower image layers take longer to locate than those in higher
-layers. However, a *copy up* operation only occurs once per file on any given
-container. Subsequent reads and writes happen against the file's copy already
-*copied-up* to the container's top layer.
+3.  If you are using a different storage driver, that means that Docker has been
```

Comment:
If you run on a vanilla kernel (without the kernel-extra package installed), the defaults will be;

- kernel < 3.16: device mapper
- kernel >= 3.16: overlay
- kernel >= 4.x overlay2